core/src/impl/Kokkos_Core.cpp

```diff
@@ -176,6 +176,10 @@ unsigned get_process_id() {
 #endif
 }
 
+bool is_valid_num_threads(int x) { return x > 0; }
+
+bool is_valid_device_id(int x) { return x >= 0; }
```

Comment:
Should there also be a check here against the upper limit of valid device ids?

Reply:
The limit is the sky